src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -512,6 +490,107 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
             return actualArguments.AsImmutableOrNull();
         }
 
+        internal static ImmutableArray<BoundExpression> MakeArgumentsInEvaluationOrder(
+            Binder binder,
+            SyntaxNode syntax,
+            ImmutableArray<BoundExpression> arguments,
+            Symbol methodOrIndexer,
+            MethodSymbol optionalParametersMethod,
+            bool expanded,
+            ImmutableArray<int> argsToParamsOpt,
+            bool invokedAsExtensionMethod,
+            /* out */ ArrayBuilder<ArgumentKind> argumentKindBuilder,
+            /* out */ ArrayBuilder<ParameterSymbol> parameterSymbolBuilder,
+            ThreeState enableCallerInfo = ThreeState.Unknown)
+        {
+            Debug.Assert(binder != null && argumentKindBuilder.IsEmpty() && parameterSymbolBuilder.IsEmpty());
+
+            // Either the methodOrIndexer is a property, in which case the method used
+            // for optional parameters is an accessor of that property (or an overridden
+            // property), or the methodOrIndexer is used for optional parameters directly.
+            Debug.Assert(((methodOrIndexer.Kind == SymbolKind.Property) && optionalParametersMethod.IsAccessor()) ||
```

Comment:
>(methodOrIndexer.Kind == SymbolKind.Property) && optionalParametersMethod.IsAccessor() [](start = 26, length = 86)

This feels strange. Does this mean that parameter symbols won't correspond to the property symbol? This doesn't sound like the behavior one would expect.